packages/xml-views-validation/src/validators/elements/unknown-tag-name.ts

```diff
@@ -0,0 +1,199 @@
+import { UnknownTagNameIssue } from "../../../api";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { XMLElement, XMLToken } from "@xml-tools/ast";
+import {
+  getUI5ClassByXMLElement,
+  ui5NodeToFQN,
+  getUI5NodeFromXMLElementNamespace,
+  getUI5AggregationByXMLElement,
+} from "@ui5-language-assistant/logic-utils";
+import {
+  getMessage,
+  UNKNOWN_CLASS_IN_NS,
+  UNKNOWN_CLASS_WITHOUT_NS,
+  UNKNOWN_AGGREGATION_IN_CLASS,
+  UNKNOWN_TAG_NAME_IN_CLASS,
+} from "../../utils/messages";
+
+export function validateUnknownTagName(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): UnknownTagNameIssue[] {
+  // Can't give an error if there is no position
+  if (xmlElement.syntax.openName == undefined || xmlElement.name === null) {
+    return [];
+  }
+
+  // To avoid false positives, we only validate tags with recognized namespaces (from the model)
+  // or without a namespace (and for tags without a namespace, we only validate if
+  // they cannot be user-defined classes or aggregations).
+
+  // Get the namespace
+  if (xmlElement.ns !== undefined) {
+    // The cast is necessary because typescript doesn't recognize the conditions above as type guards
+    return validateTagWithNamespace(xmlElement as NamedXMLElementWithNS, model);
+  } else {
+    // The cast is necessary because typescript doesn't recognize the condition above as a type guard
+    return validateTagWithoutNamespace(xmlElement as NamedXMLElement, model);
+  }
+}
+
+type NamedXMLElement = XMLElement & {
+  name: string;
+  syntax: { openName: XMLToken };
+};
+type NamedXMLElementWithNS = NamedXMLElement & { ns: string };
+
+function validateTagWithNamespace(
+  xmlElement: NamedXMLElementWithNS,
+  model: UI5SemanticModel
+): UnknownTagNameIssue[] {
+  const issueDefaults = {
+    kind: "UnknownTagName" as "UnknownTagName",
+    offsetRange: {
+      start: xmlElement.syntax.openName.startOffset,
+      end: xmlElement.syntax.openName.endOffset,
+    },
+    severity: "error" as "error",
+  };
+
+  // To avoid false positives, we assume unrecognized namespaces are user-defined so we don't validate tags in them
+  // (they could be user-defined class tags).
+  // They could also be valid special namespaces like xhtml.
+  // TODO There should be an error in xml-tools if the namespace is not defined in an xmlns attribute
+  const { namespace: ui5Namespace } = getUI5NodeFromXMLElementNamespace(
+    xmlElement,
+    model
+  );
+  if (ui5Namespace === undefined) {
+    return [];
+  }
+
+  // Check if it's a known class (aggregations don't have namespaces)
+  const ui5Class = getUI5ClassByXMLElement(xmlElement, model);
+  if (ui5Class !== undefined) {
+    return [];
+  }
+
+  return [
+    {
+      ...issueDefaults,
+      message: getMessage(
+        UNKNOWN_CLASS_IN_NS,
+        xmlElement.name,
+        ui5NodeToFQN(ui5Namespace)
```

Comment:
Should we use some "pure" xml utils to extract the Namespace
instead of relying on the FQN of a UI5Node?

Currently this assumes that if `ui5Namespace === undefined` there is no issue.
But that may not be the case in the future (e.g custom controls depending how we model them).

Reply:
We can use `xmlElement.namespaces[xmlElement.ns]`, do you think it would be better? The FQN should be exactly the same but if we find a different namespace (due to a bug or different logic) I would prefer if the user could see where we actually looked for the class.

Reply:
What about: https://github.com/SAP/ui5-language-assistant/blob/master/packages/logic-utils/api.d.ts#L14-L18 ?

Reply:
This returns the class FQN, not namespace. If we use that the message will be different (`class <FQN> was not found` instead of `class <name> was not found in <namespace>`).

Reply:
ok